aten/src/ATen/native/cuda/ScatterGatherKernel.cu

```diff
@@ -420,9 +378,9 @@ struct cuda_scatter_fill_base_kernel {
   }
 }; // struct cuda_scatter_fill_base_kernel
 
-void gather_cuda_kernel(Tensor& result, const Tensor& self, int64_t dim, const Tensor& index) {
+void gather_cuda_kernel(const Tensor& result, const Tensor& self, int64_t dim, const Tensor& index) {
   cuda_scatter_gather_base_kernel</*is_scatter_like=*/false>()(
-    result, dim, index, self,
+    const_cast<Tensor&>(result), dim, index, self,
```

Comment:
Same as in the cpu kernel: I think it's better to propagate this const instead of casting it here.